buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/versioning/BuildVersionPlugin.kt

```diff
@@ -213,3 +275,17 @@ fun Project.isPromotionBuild(): Boolean =
 private
 val Project.buildTypes
     get() = extensions.getByName<NamedDomainObjectContainer<BuildType>>("buildTypes")
+
+
+// TODO: move to ProviderFactory and make it a build logic input
+fun Project.gradleProperty(propertyName: String): Provider<Any> =
+    provider { findProperty(propertyName) }
+
+
+/**
+ * Creates a [Provider] that returns `true` when this [Provider] has a value
+ * and `false` otherwise. The returned [Provider] always has a value.
+ * @see Provider.isPresent
+ */
+fun <T> Provider<T>.presence(): Provider<Boolean> =
```

Comment:
:o: I would make this extension and the one above private until it's needed to expose them

Reply:
Right. Will do 👍